lib/buffer.js

```diff
@@ -70,6 +70,11 @@ function Buffer(arg, encoding) {
 Object.setPrototypeOf(Buffer.prototype, Uint8Array.prototype);
 Object.setPrototypeOf(Buffer, Uint8Array);
 
+if (Symbol.species && Buffer[Symbol.species] === Buffer) {
```

Comment:
In Node we probably want to wait for V8 4.9 to land and then remove the if, replacing it with a comment.